### DISCUSSION TITLE
Cannot read property 'serialize' of null

### DISCUSSION THREAD

**Discussion by brunodsazevedo** [adonisjs/core#2650] [Help] [answered]
I am making a user controller in a CRUD, where each user will have a profile. Checking in the documentation, when I do a BelongsTo or HasOne type relationship, the return should be, in my case, profile or null, but this error is returning:

`{
  "message": "Cannot read property 'serialize' of null",
  "stack": "TypeError: Cannot read property 'serialize' of null\n    at /home/bruno/projetos/wacash/backend/node_modules/@adonisjs/lucid/build/src/Orm/BaseModel/index.js:1339:25\n    at Array.reduce (<anonymous>)\n    at Proxy.serializeRelations (/home/bruno/projetos/wacash/backend/node_modules/@adonisjs/lucid/build/src/Orm/BaseModel/index.js:1316:45)\n    at Proxy.serialize (/home/bruno/projetos/wacash/backend/node_modules/@adonisjs/lucid/build/src/Orm/BaseModel/index.js:1356:21)\n    at Proxy.toJSON (/home/bruno/projetos/wacash/backend/node_modules/@adonisjs/lucid/build/src/Orm/BaseModel/index.js:1396:21)\n    at JSON.stringify (<anonymous>)\n    at Object.stringify [as default] (/home/bruno/projetos/wacash/backend/node_modules/fast-safe-stringify/index.js:14:16)\n    at Object.safeStringify (/home/bruno/projetos/wacash/backend/node_modules/@poppinss/utils/build/src/safeStringify.js:39:45)\n    at Response.writeBody (/home/bruno/projetos/wacash/backend/node_modules/@adonisjs/http-server/build/src/Response/index.js:210:31)\n    at Response.finish (/home/bruno/projetos/wacash/backend/node_modules/@adonisjs/http-server/build/src/Response/index.js:837:32)"
}`

This is my controller:

```import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
import User from 'App/Models/User'
export default class UsersController {
  public async index ({}: HttpContextContract) {
    const users = await User.query().preload('profile')
    return users
  }
// others methods
}
```

This is my migration:

```import BaseSchema from '@ioc:Adonis/Lucid/Schema'

export default class Profiles extends BaseSchema {

  protected tableName = 'profiles'
  
  public async up () {
    this.schema.createTable(this.tableName, (table) => {
      table.increments('id')
      table.integer('user_id').notNullable().unique().unsigned().references('id').inTable('users').onDelete('CASCADE')
      table.string('first_name').notNullable()
      table.string('last_name').notNullable()
      table.string('cpf', 11).notNullable().unique()
      table.string('address').notNullable()
      table.string('address_number', 10).notNullable()
      table.string('address_detail', 45).nullable()
      table.string('neighborhood').notNullable()
      table.string('city').notNullable()
      table.string('state', 2).notNullable()
      table.string('zip_code', 10).notNullable()
      table.string('phone', 13).nullable()
      table.string('cellphone', 14).notNullable()
      table.boolean('whatsapp').notNullable().defaultTo(0)
      table.text('avatar', 'longtext').nullable()
      table.timestamps(true)
    })
  }
  public async down () {
    this.schema.dropTable(this.tableName)
  }
}
```

and this is my Model:

```import { DateTime } from 'luxon'
import Hash from '@ioc:Adonis/Core/Hash'
import Profile from 'App/Models/Profile'
import {
  column,
  beforeSave,
  BaseModel,
  belongsTo,
  BelongsTo,
} from '@ioc:Adonis/Lucid/Orm'

export default class User extends BaseModel {
  @column({ isPrimary: true })
  public id: number

  @column()
  public email: string

  @column({ serializeAs: null })
  public password: string

  @column()
  public rememberMeToken?: string

  @column()
  public email_validation: boolean

  @column()
  public status: boolean

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @belongsTo(() => Profile , {
    foreignKey: 'id',
    localKey: 'user_id',
  })
  public profile: BelongsTo<typeof Profile>

  @beforeSave()
  public static async hashPassword (user: User) {
    if (user.$dirty.password) {
      user.password = await Hash.make(user.password)
    }
  }
}
```

Am I doing something wrong?

**Comment by RomainLanz** [comment, marked as answer]
Hey @brunodsazevedo! 👋 

Which version of AdonisJS Lucid are you using?
The latest one is `14.0.2`.

**Reply by brunodsazevedo** [reply]
In my package.json is `"@adonisjs/lucid": "^14.0.0",`

**Reply by brunodsazevedo** [reply]
I updated to version 14.0.2 and it worked. Thanks!

**Comment by yusuftekinx** [comment]
My problem is not due to version. Is there any other solution. it happened after using hasOne
